### ISSUE TITLE
Error: A Git repository is required for this functionality.

### ISSUE THREAD

**Issue by ogorodriguez** [workflowr/workflowr#249]
When I run this command I get:

> wflow_status()
>     Error: A Git repository is required for this functionality.

I thought that somehow I lost my Git configuration in my laptop so I tried reinstalling the Git repo again by running:

> wflow_git_config(user.name = "...", user.email = "...")
> Current settings:
>   user.email: ...@gmail.com
>   user.name: ...
> Error: Some settings already exist. Set overwrite=TRUE to overwrite them.

I do the overwrite = TRUE command  and get...

> wflow_git_config(user.name = "...", user.email = "...@gmail.com", overwrite = TRUE)
> Current settings:
>   user.email: ...@gmail.com
>   user.name: ...
> The settings above will be overwritten.
> Current Git user.name:	...
> Current Git user.email:	...@gmail.com
> Other Git settings:
> 	filter.lfs.clean:	git-lfs clean -- %f
> 	filter.lfs.process:	git-lfs filter-process
> 	filter.lfs.required:	true
> 	filter.lfs.smudge:	git-lfs smudge -- %f

I am trying to determine what may be the issue.  Is it realted to git2r as mentioned in this post? https://github.com/jdblischak/workflowr/issues/160

I have tried restarting the R session and checking any packages installed outside the project workflowr session.  Any help on the matter will be highly appreciated.  I have tried this with several of my projects and get the same error and I still have no idea of the direction I should take to resolve this.

**Comment by pcarbo** [comment]
@ogorodriguez It is also possible that your R working directory was not within your workflowr project. Please check this by running `getwd()`.

**Comment by jdblischak** [comment]
I agree that we need to first determine if this is a working directory issue or not. Please run the following commands and report the output:

```
getwd()
dir.exists(".git")
packageVersion("git2r")
git2r::repository(".", discover = TRUE)
```

**Comment by ogorodriguez** [comment]
Hello John, pcarbo,

Here's my result
> > getwd()
> [1] "C:/Users/.../Desktop/Rprojects/name-of-project"
> > dir.exists(".git")
> [1] FALSE
> > packageVersion("git2r")
> [1] ‘0.28.0’
> > git2r::repository(".", discover = TRUE)
> Error in git2r::repository(".", discover = TRUE) : 
>   The 'path' is not in a git repository

**Comment by jdblischak** [comment]
Ok, that confirms the issue is not the working directory. It's as the error says; there is no Git repository. How did you create this workflowr project? Did you use `wflow_start()`? The default setting is `git = TRUE`, which should have initialized a Git repository for the project. Has `wflow_status()` ever worked for this project?

**Comment by ogorodriguez** [comment]
Hello, I know this is a way late reply to this message.  Yes.  I used wflow_start() to create the project.  When I run wflow_status() I get:
> wflow_start()
Error in wflow_start() : argument "directory" is missing, with no default

**Comment by jdblischak** [comment]
@ogorodriguez The code chunk you copy-pasted is running `wflow_start()`. I had asked about `wflow_status()`.

From your previous reply on May 3rd, we had confirmed that your project is not a Git repository. This is a vital component of workflowr. Is your workflowr project on GitHub? Do you ever run `wflow_publish()`? Are you interested in using the version control features of workflowr for your project? I'm happy to help you continue troubleshooting, but I need some more context.

As a test case, here is some code that creates a temporary workflowr project (that will automatically be deleted when you close R). Could you please run it and copy-paste the output?

```R
library(workflowr)
x <- tempfile()
wflow_start(x, change_wd = FALSE)
git2r::in_repository(x)
```

You should see something like this:

```R
> library(workflowr)
This is workflowr version 1.6.2.9000
Run ?workflowr for help getting started
> x <- tempfile()
> wflow_start(x, change_wd = FALSE)
wflow_start:
- New directory created at /tmp/Rtmp2Qc1yD/file29b182caa09
- Project name is "file29b182caa09"
- Working directory continues to be /Users/john
- Git repo initiated at /tmp/Rtmp2Qc1yD/file29b182caa09
- Files were committed in version 69152e5
> git2r::in_repository(x)
[1] TRUE
```

**Comment by ogorodriguez** [comment]
Hello @jdblischak ,

A few comments here.  

Yes wflow_Status() has worked on this project before.  

This project is the one published in workflowr.io ([see here](https://workflowr.io/projects/github/ogorodriguez/033_purrr_learning/))

I ran wflow_status() again and got this error: 

> wflow_status()
> Error: A Git repository is required for this functionality.

This is what I get when running the command you suggested:

> library(workflowr)
> x <- tempfile()
> wflow_start(x, change_wd = FALSE)
> wflow_start:
> - New directory created at C:/Users/LUIS/AppData/Local/Temp/RtmpM9jhKQ/file4cd81e283dae
> - Project name is "file4cd81e283dae"
> - Working directory continues to be C:/Users/LUIS/Desktop/Rprojects/033_purrr_learning-master
> - Git repo initiated at C:/Users/LUIS/AppData/Local/Temp/RtmpM9jhKQ/file4cd81e283dae
> - Files were committed in version 142581b
> > git2r::in_repository(x)
> [1] TRUE

Thank you greatly for all the help (and patience)
